libretroshare/src/retroshare/rswire.h

```diff
@@ -73,26 +69,63 @@ class RsWirePlace
 
 };
 
+/************************************************************************
+ * Pulse comes in three flavours.
+ *
+ *
+ * Original Msg Pulse
+ * - Spontaneous msg, on your own group.
+ * - mPulseType = WIRE_PULSE_TYPE_ORIGINAL_MSG
+ * - Ref fields are empty.
+ *
+ * Reply to a Pulse (i.e Retweet), has two parts.
+ * as we want the retweet to reference the original, and the original to know about reply.
+ * This info will be duplicated in two msgs - but allow data to spread easier.
+ *
+ * Reply Msg Pulse, will be Top-Level Msg on Publisher's Group.
+ * - mPulseMode = WIRE_PULSE_TYPE_REPLY_MSG
+ * - Ref fields refer to Parent (InReplyTo) Msg.
+ *
+ * Reply Reference, is Child Msg of Parent Msg, on Parent Publisher's Group.
+ * - mPulseMode = WIRE_PULSE_TYPE_REPLY_REFERENCE
+ * - Ref fields refer to Reply Msg.
+ * - NB: This Msg requires Parent Msg for complete info, while other two are self-contained.
+ ***********************************************************************/
+
+#define  WIRE_PULSE_TYPE_ORIGINAL_MSG        (0x0001)
+#define  WIRE_PULSE_TYPE_REPLY_MSG           (0x0002)
+#define  WIRE_PULSE_TYPE_REPLY_REFERENCE     (0x0004)
+
+#define  WIRE_PULSE_TYPE_SENTIMENT_POSITIVE  (0x0010)
+#define  WIRE_PULSE_TYPE_SENTIMENT_NEUTRAL   (0x0020)
+#define  WIRE_PULSE_TYPE_SENTIMENT_NEGATIVE  (0x0040)
```

Comment:
Did you consider to use classed enums instead of yet another bunch of ints? see #1785

Reply:
I can change this. Do Enums work well for bitwise flags?

Reply:
Yes, there is the `RS_REGISTER_ENUM_FLAGS_TYPE()` macro that was made for exactly that use case. 
Beside having type safety, you additionally need `!!` for bool conversion when doing something like `if (!!(relayMode & RsDhtRelayMode::ENABLED))`

For reference see https://github.com/RetroShare/RetroShare/blob/master/libretroshare/src/retroshare/rsflags.h#L34 or 1b29e81e899506b4a8a87e2325e35b7a14899a82 or acda7b8cc4ce66c339b0e6f933dff67e29fba252